src/udev/udevadm-verify.c

```diff
@@ -0,0 +1,123 @@
+               "  -V --version                         Show package version\n"
+               "  -N --resolve-names=early|never       When to resolve names\n"
+               "\nSee the %s for details.\n",
+               program_invocation_short_name,
+               ansi_highlight(),
+               ansi_normal(),
+               link);
+
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+        static const struct option options[] = {
+                { "help",          no_argument,       NULL, 'h' },
+                { "version",       no_argument,       NULL, 'V' },
+                { "resolve-names", required_argument, NULL, 'N' },
+                {}
+        };
+
+        int c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "hVN:", options, NULL)) >= 0)
+                switch (c) {
+                case 'h':
+                        return help();
+                case 'V':
+                        return print_version();
+                case 'N':
+                        arg_resolve_name_timing = resolve_name_timing_from_string(optarg);
+                        if (arg_resolve_name_timing < 0)
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "--resolve-names= must be early or never");
+                        /*
+                         * In the verifier "late" has the effect of "never",
+                         * and "never" would generate irrelevant diagnostics,
+                         * so map "never" to "late".
+                         */
+                        if (arg_resolve_name_timing == RESOLVE_NAME_NEVER)
+                                arg_resolve_name_timing = RESOLVE_NAME_LATE;
+                        break;
+                case '?':
+                        return -EINVAL;
+                default:
+                        assert_not_reached();
+                }
+
+        if (optind == argc)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "No rules file specified.");
+
+        return 1;
+}
+
+static int verify_rules_file(UdevRules *rules, const char *fname) {
+        int r;
+
+        r = udev_rules_parse_file(rules, fname);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse rules file %s: %m", fname);
+
+        unsigned int issues = udev_check_current_rule_file(rules);
+        unsigned int mask = (1U << LOG_ERR) | (1U << LOG_WARNING);
+        if (issues & mask)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "%s: udev rules check failed", fname);
+
+        return EXIT_SUCCESS;
+}
+
+static int verify_rules(UdevRules *rules, unsigned int num_args, char **args) {
+        int rv = EXIT_SUCCESS;
+
+        for (unsigned int i = 0; i < num_args; ++i) {
+                int r = verify_rules_file(rules, args[i]);
```

Comment:
Please always declare the special variable `int r;` at the beginning of the function, and return 0 on success:
```c
static int verify_rules(UdevRules *rules, unsigned int num_args, char **args) {
        int r, rv = 0;

        for (...) {
                r = verify_rules_file(...);
                if (r < 0 && rv >= 0)
                        rv = r;
        }

        return rv;
}
```